include/zephyr/drivers/i2c.h

```diff
@@ -900,6 +914,57 @@ static inline int i2c_transfer_signal(const struct device *dev,
 
 #endif /* CONFIG_I2C_CALLBACK */
 
+
+#if defined(CONFIG_I2C_RTIO) || defined(DOXYGEN)
+
+/**
+ * @brief Submit request(s) to an I2C device with RTIO
+ *
+ * @param iodev_sqe Prepared submissions queue entry connected to an iodev
+ *                  defined by I2C_DT_IODEV_DEFINE.
+ */
+static inline void i2c_iodev_submit(struct rtio_iodev_sqe *iodev_sqe)
+{
+	const struct i2c_dt_spec *dt_spec = iodev_sqe->sqe->iodev->data;
+	const struct device *dev = dt_spec->bus;
+	const struct i2c_driver_api *api = (const struct i2c_driver_api *)dev->api;
+
+	api->iodev_submit(dt_spec->bus, iodev_sqe);
+}
+
+extern const struct rtio_iodev_api i2c_iodev_api;
+
+/**
+ * @brief Define an iodev for a given dt node on the bus
+ *
+ * These do not need to be shared globally but doing so
+ * will save a small amount of memory.
+ *
+ * @param node DT_NODE
+ */
+#define I2C_DT_IODEV_DEFINE(name, node_id)					\
```

Comment:
could this iodev be created as part of the device with a getter function for getting the handle so that sharing this iodev is forced?

Reply:
It could be, though the iodev is not just the i2c bus but a device on the i2c bus. So its closer to the struct i2c_dt_spec and those we create ad-hoc for each device on the bus.

This closely matches that.